docs/ref/csp.txt

```diff
@@ -0,0 +1,279 @@
+.. _content-security-policy:
+
+=============================
+Content Security Policy (CSP)
+=============================
+
+.. module:: django.middleware.csp
+   :synopsis: Middleware for Content Security Policy headers
+
+Django provides built-in support for Content Security Policy (CSP) headers through the CSP
+middleware. Content Security Policy is a powerful security feature that helps protect web applications
+against various types of attacks, particularly Cross-Site Scripting (XSS) and other code injection attacks.
+
+CSP works by allowing you to specify which sources of content are considered trusted, and instructing
+the browser to only execute or render resources from those trusted sources. This effectively creates a
+whitelist of content sources, significantly reducing the risk of malicious code execution.
+
+Some key benefits of implementing CSP include:
+
+1. Mitigating XSS attacks by preventing the execution of inline scripts and restricting which external
+   scripts can be loaded.
+2. Controlling which external resources (like images, fonts, or stylesheets) can be loaded by your site.
+3. Preventing unwanted framing of your site, protecting against clickjacking attacks.
+4. Reporting policy violations to a specified URL, allowing you to monitor potential attacks or
+   misconfigured directives.
+
+CSP provides a robust and flexible framework for securing your web application, giving you fine-grained
+control over the resources your site can use and interact with. By leveraging Django's CSP middleware,
+you can easily implement and manage these policies in your project.
```

Comment:
My 2¢ which is likely to start a flamewar 😅: One of the main criticisms I often hear about the Django docs is that too much introductory/explanatory material is put into the reference section. Experienced devs just often want to get to the guts without having to scroll/skim past explanations.

My suggestion is this sort of thing ^ could potentially be moved to the `/topics/security` section with a bit dedicated to CSP, and this part here could just be a super super brief "CSP is xxx … for more info see <link to topic>"

Reply:
I moved the intro to the `topics/security` section.